pkg/controllers/work/apply_controller.go

```diff
@@ -536,6 +544,62 @@ func trackJobAvailability(curObj *unstructured.Unstructured) (ApplyAction, error
 	return manifestNotTrackableAction, nil
 }
 
+func trackServiceAvailability(curObj *unstructured.Unstructured) (ApplyAction, error) {
+	var service v1.Service
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(curObj.Object, &service); err != nil {
+		return errorApplyAction, controller.NewUnexpectedBehaviorError(err)
```

Comment:
this is an unexpected error so there is an error log already

Reply:
The log will be like "controller/controller.go:54] "Unexpected behavior identified by the controller" err="the same cluster `dev-aks-1` has bindings `kuard-demo-dev-aks-1-04e86a3c` and `kuard-demo-dev-aks-1-22574bf0` pointing to it"

It will be logged in the NewUnexpectedBehaviorError func and sometimes it's hard to tell where the error comes from.

Reply:
does that cause redundant (back to back) logs?

Reply:
nvm, just find that the error will be logged in the https://github.com/Azure/fleet/blob/main/pkg/controllers/work/apply_controller.go#L386 which has enough info, eg, the object name and its gvr.